agent/main.go

```diff
@@ -109,6 +110,11 @@ func main() {
 
 	k8sClient := k8s.NewClient(sharedInformers)
 
+	var proxyAddr string
+	if proxyAddrOverride != nil {
+		proxyAddr = *proxyAddrOverride
+	}
+
```

Comment:
since `k8sClient` needs `proxyAddr`, how about set it in `k8s.NewClient()`, and avoid passing it through `linkerdInfoHandler`.

Reply:
that was my initial approach but it seemed a bit too specific for a k8s client. But yes, lets do that